Source/Core/InputCommon/ControllerInterface/DInput/DInputKeyboardMouse.cpp

```diff
@@ -143,51 +203,447 @@ KeyboardMouse::KeyboardMouse(const LPDIRECTINPUTDEVICE8 kb_device,
+{
+  // Sage 4/2/2022: Gameplan is to move the mouse towards the nearest gate if the distance the mouse
+  // is outside of the octagon is large enough to overcome friction.The amount the mouse will move
+  // towards the gate will be the distance outside of the octagon minus friction. This is to
+  // simulate the force required to get a real stick into gates while keeping the ability to hit the
+  // values in between gates consistently. If the distance outside of the octagon is not large
+  // enough to overcome friction, the mouse will get moved to the closest point inside of the
+  // octagon. This is almost always going to be directly on the line between octagon points. I may
+  // need to bring the mouse a tiny bit inside of the octagon instead of directly on the nearest
+  // line if the section of the code that determines whether or not the mouse is inside of the
+  // octagon struggles with the mouse_point being directly on the line.
+  long distance_to_octagon_points[8] = {0};
+  for (int i = 0; i < 8; i++)
+  {
+    distance_to_octagon_points[i] =
+        Calculate_Distance_between_Points(mouse_point, octagon_points[i]);
+  }
+
+  long min_distance = distance_to_octagon_points[0];
+  long index_of_closest_octagon_point = 0;
+  for (int i = 0; i < 8; i++)
+  {
+    if (distance_to_octagon_points[i] < min_distance)
+    {
+      min_distance = distance_to_octagon_points[i];
+      index_of_closest_octagon_point = i;
+    }
+  }
+
+  long index_of_second_line_point =
+      Find_Second_Line_Point(mouse_point, octagon_points, index_of_closest_octagon_point);
+
+  double rise = static_cast<double>(octagon_points[index_of_second_line_point].y) -
+                static_cast<double>(octagon_points[index_of_closest_octagon_point].y);
+  double run = static_cast<double>(octagon_points[index_of_second_line_point].x) -
+               static_cast<double>(octagon_points[index_of_closest_octagon_point].x);
+  double slope_of_octagon_line = rise / run;
+  double y_intercept_of_octagon_line =
+      slope_of_octagon_line * (0 - octagon_points[index_of_closest_octagon_point].x) +
+      octagon_points[index_of_closest_octagon_point].y;
+
+  rise = static_cast<double>(center_of_screen.y) - static_cast<double>(mouse_point.y);
+  run = static_cast<double>(center_of_screen.x) - static_cast<double>(mouse_point.x);
+  double slope_of_line_from_center_to_mouse = rise / run;
+  if (abs(run) < 1 /*Magic number is range around 0 where we just say the slope is zero*/)
+  {
+    if (mouse_point.y < center_of_screen.y)
+    {
+      mouse_point = octagon_points[NORTH];
+    }
+    if (mouse_point.y > center_of_screen.y)
+    {
+      mouse_point = octagon_points[SOUTH];
+    }
+    return;
+  }
+  double y_intercept_of_line_from_center_to_mouse =
+      slope_of_line_from_center_to_mouse * (0 - center_of_screen.x) + center_of_screen.y;
+
+  POINT point_of_intersection = {0};
```

Comment:
```suggestion
  POINT point_of_intersection = {};
```

`{0}` is a C-ism that can lead to missing aggregate initialization warnings in C++, empty braces are preferable

Reply:
Got it 👍